### PR TITLE
Prohibit empty strings

### LINKED ISSUE #85: Prohibit empty strings
The schema uses strings in the definition. However, it still validates successful if they are empty which could allow defeating a required information by putting in an empty string. I suggest to prohibit empty strings - if no information is present, the field should not be present as well.

### PR DESCRIPTION
**Pull request by tschmidtb51**
- resolves #85 
- add "minLength" in properties
- add generic keywords for documentation

### CODE REVIEW

File: csaf_2.0/json_schema/csaf_json_schema.json [hunk 1]
```diff
@@ -262,10 +270,12 @@
           "type": "object",
           "properties": {
             "namespace": {
-              "type": "string"
+              "type": "string",
+              "format": "uri"
```
Comment (sthagen) [inline csaf_2.0/json_schema/csaf_json_schema.json:274]: I am so used to namespaces not being URIs that I wonder if we better leave the refined type to the producer and consumer?

**Comment by santosomar** [comment]
Please try to resolve the conflicts before approving the merge of the pull request.

**Comment by tolim** [comment]
> Please try to resolve the conflicts before approving the merge of the pull request.

We had too many changes to the document, so this pull request got some conflicts. Those are removed now.

**Comment by sthagen** [comment]
Document properties like `title`, `description`, etc. of our CSAF DSL make me nervous as part of the mix into json schema language which uses these "keys" for generic meta purposes :wink:

**Pull request merged**